internal/v4/api_test.go

```diff
@@ -453,7 +453,8 @@ func (s *APISuite) TestEndpointGet(c *gc.C) {
 func (s *APISuite) TestAllMetaEndpointsTested(c *gc.C) {
 	// Make sure that we're testing all the metadata
 	// endpoints that we need to.
-	s.addCharm(c, "wordpress", newResolvedURL("~charmers/precise/wordpress-23", 23))
+	id, _ := s.addCharm(c, "wordpress", newResolvedURL("~charmers/precise/wordpress-23", 23))
```

Comment:
rather than calling setPublic after every addCharm call, maybe addCharm
itself could do that?